src/App.tsx

```diff
@@ -1,25 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [selectedPerPageValue, setSelectedPerPageValue] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const handlePerPageChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedPerPageValue(Number(event.target.value));
+    setCurrentPage(1);
+  };
+
+  const onPageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const indexOfLastItem = currentPage * selectedPerPageValue;
+  const indexOfFirstItem = indexOfLastItem - selectedPerPageValue;
+
+  const currentItems = items.slice(indexOfFirstItem, indexOfLastItem);
```

Comment:
Use the function - getNumbers immediately.
```suggestion
  const currentItems = getNumbers(indexOfFirstItem, indexOfLastItem)).map(n => `Item ${n}`;
```